code/modules/reagents/chemistry/machinery/coffeemaker.dm

```diff
@@ -0,0 +1,411 @@
+	cartridge = new /obj/item/coffee_cartridge
+
+/obj/machinery/coffeemaker/constructed/Initialize(mapload)
+	. = ..()
+	QDEL_NULL(coffeepot)
+	QDEL_NULL(cartridge)
+	update_appearance()
+
+/obj/machinery/coffeemaker/deconstruct()
+	coffeepot?.forceMove(drop_location())
+	coffeepot = null
+	cartridge?.forceMove(drop_location())
+	cartridge = null
+	return ..()
+
+/obj/machinery/coffeemaker/Destroy()
+	QDEL_NULL(coffeepot)
+	QDEL_NULL(cartridge)
+	return ..()
+
+/obj/machinery/coffeemaker/RefreshParts()
+	. = ..()
+	speed = 1
+	for(var/obj/item/stock_parts/micro_laser/laser in component_parts)
+		speed = laser.rating
+
+/obj/machinery/coffeemaker/examine(mob/user)
+	. = ..()
+	if(!in_range(user, src) && !issilicon(user) && !isobserver(user))
+		. += span_warning("You're too far away to examine [src]'s contents and display!")
+		return
+
+	if(brewing)
+		. += span_warning("\The [src] is brewing.")
+		return
+
+	if(panel_open)
+		. += span_notice("[src]'s maintenance hatch is open!")
+		return
+
+	if(coffeepot || cartridge)
+		. += span_notice("\The [src] contains:")
+		if(coffeepot)
+			. += span_notice("- \A [coffeepot].")
+		if(cartridge)
+			. += span_notice("- \A [cartridge].")
+
+	if(!(machine_stat & (NOPOWER|BROKEN)))
+		. += "[span_notice("The status display reads:")]\n"+\
+		span_notice("- Brewing coffee at <b>[speed*100]%</b>.")
+		if(coffeepot)
+			for(var/datum/reagent/consumable/cawfee in coffeepot.reagents.reagent_list)
+				. += span_notice("- [cawfee.volume] units of coffee in pot.")
+		if(cartridge)
+			if(cartridge.charges < 1)
+				. += span_notice("- grounds cartridge is empty.")
+			else
+				. += span_notice("- grounds cartridge has [cartridge.charges] charges remaining.")
+
+	if (coffee_cups > 1)
+		. += span_notice("There are [coffee_cups] cups left.")
+	else if (coffee_cups == 1)
+		. += span_notice("There is one cup left.")
+	else
+		. += span_notice("There are no cups left.")
+
+	if (sugar_packs > 1)
+		. += span_notice("There are [sugar_packs] packets of sugar left.")
+	else if (sugar_packs == 1)
+		. += span_notice("There is one packet of sugar left.")
+	else
+		. += span_notice("There is no sugar left.")
+
+	if (sweetener_packs > 1)
+		. += span_notice("There are [sweetener_packs] packets of sweetener left.")
+	else if (sweetener_packs == 1)
+		. += span_notice("There is one packet of sweetener left.")
+	else
+		. += span_notice("There is no sweetener left.")
+
+	if (creamer_packs > 1)
+		. += span_notice("There are [creamer_packs] packets of creamer left.")
+	else if (creamer_packs == 1)
+		. += span_notice("There is one packet of creamer left.")
+	else
+		. += span_notice("There is no creamer left.")
+
+/obj/machinery/coffeemaker/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(!can_interact(user) || !user.canUseTopic(src, !issilicon(user), FALSE, NO_TK))
+		return
+	if(brewing)
+		return
```

Comment:
Return `SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN` for all of these--`..()` should never be returned from the `*_secondary` procs.